### PR TITLE
Fixes for centos docker image

### PR DESCRIPTION
**Pull request by howardjohn**
As tested in https://github.com/istio/proxy/pull/2865

* Build our own gn
* Fix path
* Add gsutil for pushing artifacts

### CODE REVIEW

File: docker/build-tools/Dockerfile.centos [hunk 1]
```diff
@@ -24,6 +24,10 @@ RUN ln -s /usr/bin/cmake3 /usr/bin/cmake
 RUN echo "/opt/rh/httpd24/root/usr/lib64" > /etc/ld.so.conf.d/httpd24.conf && \
     ldconfig
 
+# We need gsutil to push/pull
+ADD https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-sdk-${GCLOUD_VERSION}-linux-x86_64.tar.gz /tmp
```
Comment (jwendell) [inline docker/build-tools/Dockerfile.centos:28]: `GCLOUD_VERSION` is undefined?

File: docker/build-tools/Dockerfile.centos [hunk 2]
```diff
@@ -34,3 +38,9 @@ RUN curl -fsSL --output ${LLVM_RELEASE}.tar.xz https://releases.llvm.org/${LLVM_
     rm ./${LLVM_RELEASE}.tar.xz && \
     echo "/opt/llvm/lib" > /etc/ld.so.conf.d/llvm.conf && \
     ldconfig
+
+ENV PATH=/opt/llvm/bin:/usr/local/google-cloud-sdk/bin:$PATH
+
+# hadolint ignore=DL3003
+RUN cd /tmp && git clone https://gn.googlesource.com/gn && \
```
Comment (jwendell) [inline docker/build-tools/Dockerfile.centos:45]: Do you have to build a specific version? master works?

Comment (howardjohn) [reply docker/build-tools/Dockerfile.centos:45]: Looking into this. @jplevyak says it should be stable but will see if we can find the correct version to pin

Comment (howardjohn) [reply docker/build-tools/Dockerfile.centos:45]: 'gn_version': 'git_revision:5ed3c9cc67b090d5e311e4bd2aba072173e82db9',


This matches the wee8  in proxy-wasm master.   It is in the DEPS in the root of the wee8 tar ball in envoy's repository_locations.bzl BTW

**Pull request merged**